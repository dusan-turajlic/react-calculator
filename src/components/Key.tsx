import React from "react";

interface KeyProps {
    value: string

    onPress?( event: React.MouseEvent, value: string ): void,

    classNames?: string[],
    colSize?: string
}

function Key( { colSize = 'col-3', classNames = [], value, onPress = () => {} }: KeyProps ) {
    return (
        <button
            className={ `column ${ colSize } ${ classNames.join( ' ' ) }` }
            onClick={ ( event: React.MouseEvent<HTMLButtonElement> ) => { onPress( event, value ) } }>
            { value }
        </button>
    );
}

export default Key;
