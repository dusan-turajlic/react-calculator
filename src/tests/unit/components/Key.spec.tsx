import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Key from '../../../components/Key';

describe( 'Key.spec', () => {
    it( 'renders correctly', function () {
        const { container } = render( <Key value='1' /> );
        expect( container ).toMatchSnapshot();
    } );

    it( 'pressing the key passes calls on click callback', () => {
        let pressCallback = jest.fn();
        const { container } = render( <Key value='1' onPress={pressCallback} /> );
        let key = container.querySelector( 'button' );
        // @ts-ignore
        fireEvent.click( key );

        expect( pressCallback ).toHaveBeenCalledWith( expect.any( Object ), '1' );
    } );
} );
