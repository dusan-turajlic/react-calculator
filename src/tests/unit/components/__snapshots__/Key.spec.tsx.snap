// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key.spec renders correctly 1`] = `
<div>
  <button
    class="column col-3 "
  >
    1
  </button>
</div>
`;
